src/services/dashboards/modules/DashboardCloneModal.vue

```diff
@@ -121,52 +122,60 @@ export default defineComponent<Props>({
             isAllValid,
         } = useFormValidator({
             name: '',
-            visibility: '',
+            viewers: '',
         }, {
-            name(value: string) { return value.trim().length ? '' : i18n.t('DASHBOARDS.FORM.REQUIRED'); },
-            visibility(value: DashboardPrivacyType) { return value.length ? '' : i18n.t('DASHBOARDS.FORM.REQUIRED'); },
+            name(value: string) {
+                if (!value.trim().length) return i18n.t('DASHBOARDS.FORM.VALIDATION_DASHBOARD_NAME_INPUT');
+                if (state.dashboardNameList.find((d) => d === value)) return i18n.t('DASHBOARDS.FORM.VALIDATION_DASHBOARD_NAME_UNIQUE');
+                return '';
+            },
+            viewers(value: DashboardViewer) { return value.length ? '' : i18n.t('DASHBOARDS.FORM.REQUIRED'); },
         });
+
+        const dashboardDetailInfoStore = useDashboardDetailInfoStore();
         const state = reactive({
             proxyVisible: props.visible,
-            filteredVisibilityList: computed(() => (props.manageDisabled ? visibilityList.filter((item) => item.name === DASHBOARD_PRIVACY_TYPE.USER) : visibilityList)),
+            filteredVisibilityList: computed(() => visibilityList),
+            isProjectDashboard: computed(() => Object.prototype.hasOwnProperty.call(props.dashboard, 'project_id')),
+            dashboardNameList: computed<string[]>(() => {
+                if (state.isProjectDashboard) {
+                    return store.state.dashboard.projectItems
+                        .filter((item) => (item.project_id === (props.dashboard as ProjectDashboardModel).project_id))
+                        .map((_item) => {
+                            if (_item.name !== props.dashboard?.name) return _item.name;
+                            return '';
+                        });
```

Comment:
suggestion: how about this?!

```suggestion
                        .filter((item) => (item.project_id === (props.dashboard as ProjectDashboardModel).project_id) && item.name !== props.dashboard.name)
                        .map((_item) => _item.name });
```

Reply:
just suggestion! you can ignore it 😅

Reply:
Oh I like it !!